packages/arb-avm-cpp/data_storage/src/datastorage.cpp

```diff
@@ -25,8 +25,14 @@
 #include <iostream>
 #include <string>
 #include <thread>
+#include <utility>
 
 DataStorage::DataStorage(const std::string& db_path) {
+    std::atomic_init(&shutting_down, false);
+    std::atomic_init(&concurrent_database_access_counter, 0);
```

Comment:
Could we initialize these as fields in the constructor, like `DataStorage::DataStorage(const std::string& db_path) : shutting_down(false) {`? I was previously unaware `std::atomic_init` existed, but I don't think it's necessary here.

Reply:
good catch